daffodil-tdml/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -437,69 +432,81 @@ abstract class TestCase(testCaseXML: NodeSeq, val parent: DFDLTestSuite)
-          val output = Channels.newChannel(os)
-          p.save(output)
-
-          val is = new java.io.ByteArrayInputStream(os.toByteArray)
-          val input = Channels.newChannel(is)
-          val compiler_ = Compiler()
-          compiler_.reload(input)
-        } else p
+  private lazy val defaultImplementations: Seq[String] = parent.defaultImplementations
+  private lazy val tcImplementations = (testCaseXML \ "@implementations").text
+  private lazy val implementationStrings =
+    if (tcImplementations == "") defaultImplementations
+    else tcImplementations.split("""\s+""").toSeq
 
-      Right((diags, dp))
+  protected def toss(t: Throwable, implString: Option[String]) = {
+    t match {
+      case e: TDMLException => throw e
+      case e: UnsuppressableException => throw e
+      case s: scala.util.control.ControlThrowable => throw s
+      case _ => throw TDMLException(t, implString)
     }
   }
 
-  private def compileProcessor(schemaSource: DaffodilSchemaSource, useSerializedProcessor: Boolean): DFDLTestSuite.CompileResult = {
-    val pf = compiler.compileSource(schemaSource)
-    val diags = pf.getDiagnostics
-    if (pf.isError) {
-      Left(diags) // throw new TDMLException(diags)
-    } else {
-      val res = this.generateProcessor(pf, useSerializedProcessor)
-      res
-    }
-  }
+  lazy val tdmlDFDLProcessorFactory: AbstractTDMLDFDLProcessorFactory = {
+    import scala.language.reflectiveCalls
+    import scala.language.existentials
 
-  final protected def getProcessor(schemaSource: DaffodilSchemaSource, useSerializedProcessor: Boolean): DFDLTestSuite.CompileResult = {
-    val res: DFDLTestSuite.CompileResult = schemaSource match {
-      case uss: URISchemaSource =>
-        SchemaDataProcessorCache.compileAndCache(uss, useSerializedProcessor,
-          parent.checkAllTopLevel,
-          root,
-          null) {
-            compileProcessor(uss, useSerializedProcessor)
-          }
-      case _ => {
-        compileProcessor(schemaSource, useSerializedProcessor)
-      }
-    }
+    val className = "org.apache.daffodil.tdml.processor.TDMLDFDLProcessorFactory"
+
+    //
+    // If you haven't seen it before. Check out this Try(...) idiom.
+    // Much cleaner than the messy nest of individual try/catches for each case.
+    //
+    // We're not catching anything here, but we could have surrounded the tryInstance.get call with a
+    // single tier of catch, with cases for all the various throws that could have happened anywhere on
+    // the three lines of actions that can throw various things.
+    //
+    // of course this will allocate an object, so not for tightest inner loops, but the code
+    // cleanup is substantial.
+    //
+    val clazz = Try(Class.forName(className))
+    val constructor = clazz.map { _.getDeclaredConstructor() }
+    val tryInstance = constructor.map { _.newInstance().asInstanceOf[AbstractTDMLDFDLProcessorFactory] }
+    val res = tryInstance.get
```

Comment:
If no TDMLDFDLProcessorFactory is on the classpath this ``.get`` will fail with a ClassNotFoundException. Might be more helpful to instead throw a TDMLException with a helpful error message?

Reply:
fixed